packages/snaps-controllers/src/snaps/SnapController.ts

```diff
@@ -1214,9 +1226,13 @@ export class SnapController extends BaseController<
       );
     }
 
-    const isAllowlistingRequired = Object.keys(snapInfo.permissions).some(
-      (permission) => !ALLOWED_PERMISSIONS.includes(permission),
-    );
+    const isAllowlistingRequired =
+      Object.keys(snapInfo.initialPermissions).some(
+        (permission) => !ALLOWED_PERMISSIONS.includes(permission),
+      ) ||
+      Object.keys(snapInfo.dynamicPermissions || {}).some(
```

Comment:
```suggestion
      Object.keys(snapInfo.dynamicPermissions ?? {}).some(
```
Shouldn't we do this instead?

Reply:
Yes, updated.